### ISSUE TITLE
Uncaught TypeError: Cannot read property 'delete' of undefined

### ISSUE THREAD

**Issue by etorreborre** [atom-haskell/haskell-ghc-mod#14]
[Enter steps to reproduce below:]

1. ...
2. ...

**Atom Version**: 0.201.0
**System**: Mac OS X 10.10.2
**Thrown From**: [haskell-ghc-mod](https://github.com/atom-haskell/haskell-ghc-mod) package, v0.6.1

### Stack Trace

Uncaught TypeError: Cannot read property 'delete' of undefined

```
At /Users/etorreborre/.atom/packages/haskell-ghc-mod/lib/ghc-modi-process.coffee:49

TypeError: Cannot read property 'delete' of undefined
  at ChildProcess.<anonymous> (/Users/etorreborre/.atom/packages/haskell-ghc-mod/lib/ghc-modi-process.coffee:49:18)
  at emitTwo (events.js:87:13)
  at ChildProcess.emit (events.js:169:7)
  at Process.ChildProcess._handle.onexit (child_process.js:1057:12)

```

### Commands

```
     -2:52.1.0 core:backspace (ul.list-inline.tab-bar.inset-panel)
     -2:51.4.0 core:save (ul.list-inline.tab-bar.inset-panel)
     -2:51.4.0 ide-haskell:check-file (atom-text-editor.editor)
     -2:51.4.0 ide-haskell:lint-file (atom-text-editor.editor)
     -2:38.1.0 core:copy (atom-text-editor.editor.is-focused)
```

### Config

```json
{
  "core": {
    "themes": [
      "atom-light-ui",
      "atom-light-syntax"
    ]
  },
  "haskell-ghc-mod": {
    "ghcModPath": "/Users/etorreborre/projects/haskell/ghc-mod/.cabal-sandbox/bin/ghc-mod",
    "ghcModiPath": "/Users/etorreborre/projects/haskell/ghc-mod/.cabal-sandbox/bin/ghc-modi"
  }
}
```

### Installed Packages

```coffee
# User
autocomplete-haskell, v0.4.1
haskell-ghc-mod, v0.6.1
ide-haskell, v0.5.5
language-haskell, v1.0.0
language-scala, v1.1.0
linter-hlint, v0.3.5

# Dev
No dev packages
```

**Comment by TaylorScanlon** [comment]
Can you post your .cabal file? I had this issue and it was caused by not having 'main-is' defined under the executable.  Your issue may be similar.

**Comment by lierdakil** [comment]
Hello. Thank you for this report. However, this should be already fixed in v0.6.1. Could you try restarting Atom and reproducing?

Anyway, it seems like ghc-modi crashes. This usually means your Cabal project is misconfigured in some way. Dev console might have some info on what exactly happens.

**Comment by lierdakil** [comment]
Nope, sorry, scratch that, I forgot to publish latest version. 0.6.2 should work.
